receiver/googlecloudpubsubreceiver/README.md

```diff
@@ -0,0 +1,63 @@
+# Google Pubsub Receiver
```

Comment:
Hi Alexvanboxel, nice to meet you! I'm a member of the Cloud Trace team at Google. 

Cloud Trace supports the single receiver approach, with one caveat. Our public documentation will still instruct customers to create one PubSub topic per telemetry type. The multi-topic approach allows for a more fine-grained authorization model (whereby Trace can only publish to the trace topic, Logging can only publish to the logging topic, and so on). However, this shouldn't affect your implementation. You can still write a single receiver that has the ability to accept all telemetry types. Our public documentation can simply encourage "best" security practices. There would be no inherent technical limitation that would prevent the customer from ignoring our documentation and piping all telemetry types through a single PubSub topic anyways.

We also favor this [CloudEvents](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#event-data) approach (as opposed to the OneOf() approach discussed in [PR #157](https://github.com/open-telemetry/oteps/pull/157/files)), implemented as [Pub/Sub attributes](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage). Our reasoning is that customers should be able to [filter by telemetry type](https://cloud.google.com/pubsub/docs/filtering) using message attributes before needing to deserialize the message.

One question I had, though, is your reasoning for using the CloudEvents spec? I'm unfamiliar with the spec, but reading through it, it appears rather thorough. Is the expectation that the receiver (and Trace, the publisher) will be fully compliant with the spec? And if not, and we are only using portions of it for inspiration, then does it make sense to give the PubSub attributes more familiar names such as (just as an example):

- "payload_format":"binaryproto"
- "payload_message":"[opentelemetry.proto.trace.v1.ResourceSpans](https://github.com/open-telemetry/opentelemetry-proto/blob/f3b0ee0861d304f8f3126686ba9b01c106069cb0/opentelemetry/proto/trace/v1/trace.proto#L28)"

Finally, I wanted to thank you for your contributions! Let me know if you have any other questions for the Trace team.

Thanks,
Matt

Reply:
Hey, Matt: it's maybe worth having a meeting, last time I had a meeting with "Eyamba Ita" there was no mention on OTLP over Pubsub. You can mail me at "alex.vanboxel", at the company I work for "collibra.com". <- email spam bot proof